test/e2e/sagemaker/tests/test_transformjob.py

```diff
@@ -0,0 +1,154 @@
+    #logging.debug(transformjob)
+
+    # Create the k8s resource
+    reference = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL, resource_name, namespace="default")
+    resource = k8s.create_custom_resource(reference, transformjob)
+    resource = k8s.wait_resource_consumed_by_controller(reference)
+
+    assert resource is not None
+
+    yield (reference, resource)    
+
+
+@service_marker
+@pytest.mark.canary
+class TestTransformJob:
+    def _get_created_transformjob_status_list(self):
+        return ["InProgress", "Completed"]
+    
+    def _get_stopped_transformjob_status_list(self):
+        return ["Stopped", "Stopping"]
+
+    def _get_resource_transformjob_arn(self, resource: Dict):
+        assert 'ackResourceMetadata' in resource['status'] and \
+            'arn' in resource['status']['ackResourceMetadata']
+        return resource['status']['ackResourceMetadata']['arn']
+
+    def _get_sagemaker_transformjob_arn(self, sagemaker_client, transformjob_name: str):
+        try:
+            transformjob = sagemaker_client.describe_transform_job(TransformJobName=transformjob_name)
+            return transformjob['TransformJobArn']
+        except BaseException:
+            logging.error(
+                f"SageMaker could not find a transformJob with the name {transformjob_name}")
+            return None
+
+    def _get_sagemaker_transformjob_status(self, sagemaker_client, transformjob_name: str):
+        try:
+            transformjob = sagemaker_client.describe_transform_job(TransformJobName=transformjob_name)
+            return transformjob['TransformJobStatus']
+        except BaseException:
+            logging.error(
+                f"SageMaker could not find a transformJob with the name {transformjob_name}")
+            return None
+
+    def test_create_transformjob(self, xgboost_transformjob):
+        (reference, resource) = xgboost_transformjob
+        assert k8s.get_resource_exists(reference)
+
+    def test_transformjob_has_correct_arn(self, sagemaker_client, xgboost_transformjob):
+        (reference, _) = xgboost_transformjob
+        resource = k8s.get_resource(reference)
+        transformjob_name = resource['spec'].get('transformJobName', None)
+
+        assert transformjob_name is not None
+
+        resource_transformjob_arn = self._get_resource_transformjob_arn(resource)
+        expected_transformjob_arn = self._get_sagemaker_transformjob_arn(
+            sagemaker_client, transformjob_name)
+
+        assert resource_transformjob_arn == expected_transformjob_arn
+
+    def test_transformjob_has_created_status(self, sagemaker_client, xgboost_transformjob):
+        (reference, _) = xgboost_transformjob
+        resource = k8s.get_resource(reference)
+        transformjob_name = resource['spec'].get('transformJobName', None)
+
+        assert transformjob_name is not None
+
+        current_transformjob_status = self._get_sagemaker_transformjob_status(
```

Comment:
nit: should we create temp variables if they are only used once?


same for arn test

Reply:
Fixed